apicast/src/configuration.lua

```diff
@@ -66,13 +66,16 @@ local function check_rule(req, rule, usage_t, matched_rules, params)
 end
 
 local function get_auth_params(method)
-  local params
+  local params = ngx.req.get_uri_args()
 
-  if method == "GET" then
-    params = ngx.req.get_uri_args()
-  else
+  if method == "POST" then
     ngx.req.read_body()
-    params = ngx.req.get_post_args()
+    local body_params = ngx.req.get_post_args()
+
+    -- This will overwrite uri params if they appear in the request body. It's
+    -- an arbitrary decision. Overwriting the other way around would be equally
+    -- valid.
+    for k, v in pairs(body_params) do params[k] = v end
```

Comment:
I'd suggest you do the reverse: iterarte `params` and overwrite `body_params[k]`  only if `body_params[k]` is nil and return the `body_params` table.

I am assuming that `POST` requests with URI parameters are not the norm (and then probably a few parameters only). IOW, I'd expect most cases to have the size of `params` smaller or way smaller than the size of `body_params`. So as is you start with a table, `body_params`, that is likely to be very similar to what you are going to return, but you are filling up another table that is likely to be empty or mostly empty. You will need `O(2*Body+Uri)` space (and potential reallocation) to fill in `params`, whereas you will need `O(2*Uri+Body)` space to fill in `body_params`. You will also need `O(Body)` lookups into `params` vs `O(Uri)` lookups in `body_params`. If the assumption `Body >> Uri` holds, it would make sense to do the reverse.

Reply:
That is a good idea @unleashed 👍

Reply:
That's a good point @unleashed . I'll change this.

Reply:
Changed.